### DISCUSSION TITLE
Cloudflare R2 instead of S3 storage?

### DISCUSSION THREAD

**Discussion by janzheng** [pocketbase/pocketbase#145] [Q&A] [answered]
Hi everyone,

I'm trying to connect Pocketbase to R2 but I'm having trouble. For the endpoint, I'm getting "Must be a valid IP address or DNS name." but Pocketbase endpoints are in the form of `https://bucketid.r2.cloudflarestorage.com/bucketname`. 

Has anyone else had success setting up R2 as storage for Pocketbase?
Thnx!

**Comment by ganigeorgiev** [comment, marked as answer]
URLs support for the endpoint was added in v0.2.4 (https://github.com/pocketbase/pocketbase/issues/78). You can try downloading a newer version from the releases and just replace your binary (https://github.com/pocketbase/pocketbase/releases).

I haven't tested Cloudflare R2 with PocketBase but it may not be compatible since [it doesn't implement all S3 apis](https://developers.cloudflare.com/r2/platform/s3-compatibility/api/).

**Comment by MikeChongCan** [comment]
I started to use R2 just now, uploading images and videos to my bucket.

One caveat is that the region is `auto` according to their doc.

Everything works for me as of now.

**Comment by tarorg** [comment]
does cloud flare R2 works with supabase ?

**Reply by jlak0** [reply]
https://github.com/supabase/supabase
